independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/UnusedBeans.java

```diff
@@ -84,7 +83,7 @@ static Set<BeanInfo> findRemovableBeans(BeanResolver beanResolver, Collection<Be
             // Custom exclusions
             for (Predicate<BeanInfo> exclusion : allUnusedExclusions) {
                 if (exclusion.test(bean)) {
-                    LOG.debugf("Unremovable - excluded by %s: %s", exclusion.toString(), bean);
+                    LOG.debugf("Unremovable - excluded by %s: %s", exclusion, bean);
```

Comment:
Just to make sure I understand right: this relies upon the logger (not sure which component of the logger exactly, but I hope you understand) to call `toString()` on the passed object when it determines debug logging is enabled and the message is formatted?

Clever!